web/source/dom/uiTouchHandlerBase.ts

```diff
@@ -425,7 +425,11 @@ namespace com.keyman.dom {
         // Cancel touch if moved up and off keyboard, unless popup keys visible
       } else {
         let base = this.baseElement;
-        let top = (base.offsetParent as HTMLElement).offsetTop + base.offsetTop;
+        let parent = this.baseElement.parentElement;
```

Comment:
```suggestion
        let parent = base.parentElement;
```

But [`offsetParent` and `parentElement` are not the same](https://stackoverflow.com/a/17975012/1836776) so I don't think this change is going to be correct. It may be worth considering using [`Element.getBoundingClientRect`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect) as an alternative?

Reply:
I tested it as-is in the following situations:

- Emulation in Chrome, KMW's 'native' mode:
    - "Moto G4" (Android) emulation
    - iPhone SE (iOS) emulation
- 'native' mode on an iPhone SE 2
- embedded within the iOS app (via Xcode's Simulator)

Seems to work flawlessly in all cases listed above.  While I agree that the approach won't work for literally all uses of `offsetParent` that may exist, it _*does*_ appear to work for _this_ use quite well.  The OSK `_Box` element literally is the `offsetParent`... except when browsers decide `offsetParent` should return `null` instead b/c of current rendering / page-state conditions.

Reply:
It might break if used by the element hierarchy returned by the `buildDocumentationKeyboard` method (used for keyboard help)... except that 'keyboard' is explicitly non-interactive, meaning key previews don't exist there - and thus that there's nothing to break there.

Reply:
I see.

```suggestion
        let parent = base.parentElement; // parentElement is _Box
```